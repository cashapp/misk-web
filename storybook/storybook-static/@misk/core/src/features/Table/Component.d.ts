export interface ITableProps {
    data: any[];
    range?: number[];
}
export declare const Table: (props: ITableProps) => JSX.Element;
