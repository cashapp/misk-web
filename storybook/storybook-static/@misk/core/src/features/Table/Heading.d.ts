export declare const Heading: (props: {
    data: any;
}) => JSX.Element;
