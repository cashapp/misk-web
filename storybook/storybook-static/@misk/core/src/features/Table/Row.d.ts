export declare const Row: (props: {
    data: any;
    index?: number;
}) => JSX.Element;
