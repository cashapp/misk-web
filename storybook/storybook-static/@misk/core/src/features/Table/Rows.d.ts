export declare const Rows: (props: {
    data: any;
    range: number[];
}) => JSX.Element;
