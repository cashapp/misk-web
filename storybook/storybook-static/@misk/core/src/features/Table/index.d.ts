export * from "./Component";
export * from "./Heading";
export * from "./Row";
export * from "./Rows";
