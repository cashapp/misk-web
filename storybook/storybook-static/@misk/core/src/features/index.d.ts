export * from "./Navbar";
export * from "./Table";
