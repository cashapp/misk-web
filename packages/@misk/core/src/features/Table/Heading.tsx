import * as React from "react"

export const Heading = (props: { data: any }) => {
  const { data } = props
  return (
    <thead>
      <tr>
        <th key={"index"}>{""}</th>
        {Object.entries(data).map(([k]) => (
          <th key={k}>{k}</th>
        ))}
      </tr>
    </thead>
  )
}
