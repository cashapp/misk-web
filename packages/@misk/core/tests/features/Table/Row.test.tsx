import React from "react"
import { cleanup, render } from "@testing-library/react"
import { Row } from "src/features/Table"
import { cars } from "./testFixtures"

describe("Table Row", () => {
  afterEach(cleanup)
  it("Row can render", () => {
    const { asFragment } = render(<Row data={cars[0]} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
