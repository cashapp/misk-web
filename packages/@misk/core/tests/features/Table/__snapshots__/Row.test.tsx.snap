// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row Row can render 1`] = `
<DocumentFragment>
  <tr>
    <td>
      26f7ab1f-4060-497c-9a7e-a772c2bc1e50
    </td>
    <td>
      Buick
    </td>
    <td>
      LaCrosse
    </td>
    <td>
      2011
    </td>
    <td>
      41
    </td>
    <td>
      â‚¬21530,26
    </td>
    <td>
      Integrated even-keeled array
    </td>
    <td>
      41.3006835
    </td>
    <td>
      -8.4787949
    </td>
    <td>
      4/17/2018
    </td>
    <td>
      3:19:26.000
    </td>
    <td>
      6:44:28.000
    </td>
  </tr>
</DocumentFragment>
`;
